numba/rewrites/static_getitem.py

```diff
@@ -46,6 +46,48 @@ def apply(self):
         return new_block
 
 
+@register_rewrite('after-inference')
+class RewriteLiteralGetitems(Rewrite):
+    """
+    Rewrite IR expressions of the kind `getitem(value=arr, index=$XX)`
+    where `$XX` is a Literal value as
+    `static_getitem(value=arr, index=<literal value>)`.
+    """
+
+    def match(self, func_ir, block, typemap, calltypes):
+        self.getitems = getitems = {}
+        self.block = block
+        # Detect all getitem expressions and find which ones can be
+        # rewritten
+        for expr in block.find_exprs(op='getitem'):
+            if expr.op == 'getitem':
+                if isinstance(typemap[expr.index.name], types.StringLiteral):
+                    literal_value = typemap[expr.index.name].literal_value
+                    getitems[expr] = (expr.index, literal_value)
+
+        return len(getitems) > 0
+
+    def apply(self):
+        """
+        Rewrite all matching getitems as static_getitems.
+        """
+        new_block = self.block.copy()
+        new_block.clear()
+        for inst in self.block.body:
+            if isinstance(inst, ir.Assign):
+                expr = inst.value
+                if expr in self.getitems:
+                    const, lit_val = self.getitems[expr]
+                    new_expr = ir.Expr.static_getitem(value=expr.value,
+                                                      index=lit_val,
+                                                      index_var=expr.index,
+                                                      loc=expr.loc)
+                    inst = ir.Assign(value=new_expr, target=inst.target,
+                                     loc=inst.loc)
+            new_block.append(inst)
+        return new_block
```

Comment:
Given this is very similar to https://github.com/numba/numba/pull/5182/files#diff-0a5d74255d34e5c718ccf0fd86699968R6-R45 I'm wondering if they can be merged and/or if `func_ir.infer_constant()` could be made to resolve literals into consts (which I'm guessing it's not doing?!)?